gateways/istio-egress/templates/autoscale.yaml

```diff
@@ -1,4 +1,5 @@
-{{- if .Values.autoscaleMin }}
+{{ $gateway := index .Values "gateways" "istio-egressgateway" }}
```

Comment:
Go template cannot have hyphen so we need this hack

Reply:
Where would the hyphen be ? We use same thing for ingress and seems to work. 

We can make some changes - it is a new installer after all. It is a bit different from 1.0 to 1.1 in-place upgrade - the move from 1.1 to 1.2 should not have the same risks of accidentally breaking stuff, and since we move to different namespaces it is quite fine to change some names.

Reply:
Before we would set something like `.Values.something`, which a user would assign with `---set gateways.istio-egressgateway.something=whatever`. Now we need to address this at `.Values.gateways.istio-egressgateway.something` in the templates, which has the hyphen that breaks it. 

So we don't use the same for egress, as it doesn't have a prefix for values yet, which breaks everyones values.yaml files